File: query-service/src/main/java/ru/telemetry/query/model/DeviceEvent.java
package ru.telemetry.query.model;

import java.time.Instant;

/** Точка журнала событий — ответ GET /devices/events. type: "OFFLINE"/"ONLINE". Вычисляется на
 * лету из разрывов в telemetry_history (DeviceQueryService.findEvents), не хранится отдельно. */
public record DeviceEvent(String deviceId, String deviceName, String type, Instant occurredAt) {
}

File: dashboard/src/app/models/device-event.ts
// Контракт ответа query-service (GET /devices/events) — ru.telemetry.query.model.DeviceEvent.
export interface DeviceEvent {
  deviceId: string;
  deviceName: string;
  type: 'OFFLINE' | 'ONLINE';
  occurredAt: string;
}
